src/shared/cards/Card Template.js

```diff
@@ -0,0 +1,73 @@
+import Card from 'cards/Card';
+import { Set } from 'immutable';
+
+export default class CardName extends Card {
+  name = 'Card Name';
+  cost = 4;
+  VP = 3;
+  coin = 2;
+  types = new Set(['Action', 'Treasure', 'Attack', 'Reaction', 'Victory']);
+  async onPlay(player) {
+    // Vanilla Effects
+    player.draw(1);
+    player.actions++;
+    player.buys++;
+    player.money++;
+
+    // Trash card(s) in Hand, is it game.currentPlayer, or just Player?
+    const cards = await game.currentPlayer.selectCards(min, max, predicate);
+    for card in cards {
+      player.trash(card);
+    }
+
+    // Trash this card
+    player.trash(this);
+
+    // Draw card and do something if()
+    card = player.deck.last();
+    if (card.types.has('Victory')) {  // look at top card in deck, discard if it is a victory card
+      player.moveCard(this, player.deck, player.discard);
+    }
+
+    // Draw multiple cards and do something if(), basically emulating library
+    const cards = player.setAside(min, max);
+    for card in cards {
+      if card.types.has('Action') {
+        await choice = player.chooseOption(['hand', 'discard']);
+        if (choice == 'hard') {
+          player.moveCard(card, player.setAside, player.hand);
+        }
+        else {player.moveCard(card, player.setAside, player.discard);}
+      }
+    }
+
+    // Oponents draw card(s) and you do something , basically thief
+    await game.forEachOtherPlayer(async player => {
+      cards = player.deck[end-x:end];
```

Comment:
thats not a thing

Reply:
`player.deck.spliceLast(x)`
see `Pile.js`

Reply:
which actually modifies deck

Reply:
Am i redefining player in here?
player.draw(1) refers to the current player i think right?
So anything with other players should have a different variable name

Reply:
Probably don't want to actually use splice, just move cards to set aside location and then do stuff